.github/workflows/build.yml

```diff
@@ -0,0 +1,57 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    container: ghcr.io/zeldaret/ss-build:main
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [SOUE01]
+
+    steps:
+    # Checkout the repository (shallow clone)
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    # Set Git config
+    - name: Git config
+      run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+    # Copy the original files to the workspace
+    - name: Prepare
+      run: cp -R /orig .
+
+    # Build the project
+    - name: Build
+      run: |
+        python configure.py --map --version ${{ matrix.version }} \
+            --binutils /binutils --compilers /compilers
+        ninja all_source build/${{ matrix.version }}/progress.json
```

Comment:
This wasn't explicitly mentioned in the PR description but this will fail the build if any of the source files, including NonMatching ones, don't build? I'm in favor but this might affect some of the files currently in the repo.

Reply:
Ah yes, good catch, I forgot to check if ss currently builds with `all_source`. Hopefully it's not too big of a lift if not.